tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -128,36 +203,55 @@ def test_it_reports_lti_launches(self, context, pyramid_request, LtiLaunches):
         )
 
     def test_it_calls_grading_info_upsert(
-        self, context, pyramid_request, grading_info_service
+        self, context, pyramid_request, grading_info_service, view_caller
     ):
-        self.make_request(context, pyramid_request)
+        view_caller(context, pyramid_request)
 
         grading_info_service.upsert_from_request.assert_called_once_with(
             pyramid_request, h_user=context.h_user, lti_user=pyramid_request.lti_user
         )
 
     def test_it_does_not_call_grading_info_upsert_if_instructor(
-        self, context, pyramid_request, grading_info_service
+        self, context, pyramid_request, grading_info_service, view_caller
     ):
         pyramid_request.lti_user = LTIUser("USER_ID", "OAUTH_STUFF", roles="instructor")
 
-        self.make_request(context, pyramid_request)
+        view_caller(context, pyramid_request)
 
         grading_info_service.upsert_from_request.assert_not_called()
 
     def test_it_does_not_call_grading_info_upsert_if_canvas(
-        self, context, pyramid_request, grading_info_service
+        self, context, pyramid_request, grading_info_service, view_caller
     ):
         pyramid_request.params["tool_consumer_info_product_family_code"] = "canvas"
 
-        self.make_request(context, pyramid_request)
+        view_caller(context, pyramid_request)
 
         grading_info_service.upsert_from_request.assert_not_called()
 
+    @pytest.fixture(
+        params=[
+            canvas_file_basic_lti_launch_caller,
+            db_configured_basic_lti_launch_caller,
+            url_configured_basic_lti_launch_caller,
+            configure_module_item_caller,
+        ]
+    )
+    def view_caller(self, request):
+        """
+        Return a function that calls the view method to be tested.
 
-class TestCanvasFileBasicLTILaunch(ConfiguredLaunch):
+        This is a parametrized fixture. A test that uses this fixture will be
+        run multiple times, once for each parametrized version of this fixture.
+
+        See https://docs.pytest.org/en/latest/fixture.html#parametrizing-fixtures
+        """
+        return request.param
```

Comment:
`request` here is a pytest thing FYI, not a Pyramid request object. This is why our `pyramid_request` fixture has that awkward name instead of just `request`: pytest already has a builtin fixture named `request` which it uses to pass in fixture params and various other things.